src/Gyro.cpp

```diff
@@ -34,67 +42,88 @@ void Gyro::startup()
+    Wire.write(x);
+    Wire.endTransmission();
+
+    // Execute 1 byte write to MPU6050_REG_CONFIG to set slowest refresh rate
     Wire.beginTransmission(MPU6050_I2C_ADDR);
     Wire.write(MPU6050_REG_CONFIG);
-    Wire.write(6);      // 5Hz bandwidth (lowest) for smoothing
+    Wire.write(6); // 5Hz bandwidth (lowest) for smoothing
     Wire.endTransmission(true);
 
     LOGV1(DEBUG_INFO, F("GYRO:: Started"));
 }
 
+// Shuts down the MPU-6050 device.
+// Currently does nothing.
 void Gyro::shutdown()
-/* Shuts down the MPU-6050 device.
-   Currently does nothing.
-*/
 {
     LOGV1(DEBUG_INFO, F("GYRO: Shutdown"));
     // Nothing to do
 }
 
+// Returns roll & tilt angles from MPU-6050 device in angle_t object in degrees.
+// If MPU-6050 is not found then returns {0,0}.
 angle_t Gyro::getCurrentAngles()
-/* Returns roll & tilt angles from MPU-6050 device in angle_t object in degrees.
-   If MPU-6050 is not found then returns {0,0}.
-*/
 {
-    const int windowSize = 16;
+    const int windowSize = 8;
     // Read the accelerometer data
     struct angle_t result;
     result.pitchAngle = 0;
     result.rollAngle = 0;
     if (!isPresent)
-        return result;     // Gyro is not available
+        return result; // Gyro is not available
 
     for (int i = 0; i < windowSize; i++)
     {
         // Execute 6 byte read from MPU6050_REG_WHO_AM_I
         Wire.beginTransmission(MPU6050_I2C_ADDR);
-        Wire.write(MPU6050_REG_ACCEL_XOUT_H); 
+        Wire.write(MPU6050_REG_ACCEL_XOUT_H);
         Wire.endTransmission(false);
-        Wire.requestFrom(MPU6050_I2C_ADDR, 6, 1);  // Read 6 registers total, each axis value is stored in 2 registers
-        int16_t AcX = Wire.read() << 8 | Wire.read(); // X-axis value
-        int16_t AcY = Wire.read() << 8 | Wire.read(); // Y-axis value
-        int16_t AcZ = Wire.read() << 8 | Wire.read(); // Z-axis value
+        Wire.requestFrom(MPU6050_I2C_ADDR, 6, 1);     // Read 6 registers total, each axis value is stored in 2 registers
+        uint8_t b0 =  Wire.read();
+        uint8_t b1 =  Wire.read();
+        uint8_t b2 =  Wire.read();
+        uint8_t b3 =  Wire.read();
+        uint8_t b4 =  Wire.read();
+        uint8_t b5 =  Wire.read();
+        
+        int16_t AcX = b0 << 8 | b1; // X-axis value
+        int16_t AcY = b2 << 8 | b3; // Y-axis value
+        int16_t AcZ = b4 << 8 | b5; // Z-axis value
+
+        //LOGV7(DEBUG_INFO, F("GYRO: Values : %d, %d   %d, %d   %d, %d"), b0,b1,b2,b3,b4,b5);
 
         // Calculating the Pitch angle (rotation around Y-axis)
         result.pitchAngle += ((atanf(-1 * AcX / sqrtf(powf(AcY, 2) + powf(AcZ, 2))) * 180.0f / static_cast<float>(PI)) * 2.0f) / 2.0f;
         // Calculating the Roll angle (rotation around X-axis)
         result.rollAngle += ((atanf(-1 * AcY / sqrtf(powf(AcX, 2) + powf(AcZ, 2))) * 180.0f / static_cast<float>(PI)) * 2.0f) / 2.0f;
 
-        delay(10);  // Decorrelate measurements
+        delay(5); // Decorrelate measurements
```

Comment:
Why change the update rate + window size here? In the initialization you set 5Hz bandwidth but here you're checking at 200Hz (1/(5ms) == 200Hz). Am I missing something?

Reply:
Not sure. Maybe I'm not understanding how this works.

Reply:
Fixed.